client/src/components/aggregations/AggregationWidgetContainer.js

```diff
@@ -0,0 +1,215 @@
+
+const DEFAULT_AGGREGATION_WIDGET_PER_FIELD_TYPE = {
+  [CUSTOM_FIELD_TYPE.TEXT]: AGGERGATION_WIDGET_TYPE.DEFAULT,
+  [CUSTOM_FIELD_TYPE.NUMBER]: AGGERGATION_WIDGET_TYPE.IQR_BOX_PLOT,
+  [CUSTOM_FIELD_TYPE.DATE]: AGGERGATION_WIDGET_TYPE.CALENDAR,
+  [CUSTOM_FIELD_TYPE.DATETIME]: AGGERGATION_WIDGET_TYPE.CALENDAR,
+  [CUSTOM_FIELD_TYPE.ENUM]: AGGERGATION_WIDGET_TYPE.PIE,
+  [CUSTOM_FIELD_TYPE.ENUMSET]: AGGERGATION_WIDGET_TYPE.PIE,
+  [CUSTOM_FIELD_TYPE.SPECIAL_FIELD]: {
+    [SPECIAL_WIDGET_TYPES.LIKERT_SCALE]:
+      AGGERGATION_WIDGET_TYPE.LIKERT_SCALE_AND_PIE,
+    [SPECIAL_WIDGET_TYPES.RICH_TEXT_EDITOR]: AGGERGATION_WIDGET_TYPE.DEFAULT
+  }
+}
+
+export const AGGREGATION_TYPE = {
+  REPORTS_BY_TASK: "countReportsByTask",
+  COUNT_PER_DATE: "countPerDate",
+  COUNT_PER_VALUE: "countPerValue",
+  NUMBERS_LIST: "numbersList",
+  VALUES_LIST: "valuesList",
+  OBJECTS_LIST: "objectsList",
+  LIKERT_SCALE_AND_PIE_AGG: "likertScaleAndPieAgg"
+}
+
+const DEFAULT_AGGREGATION_TYPE_PER_WIDGET_TYPE = {
+  [AGGERGATION_WIDGET_TYPE.LIKERT_SCALE]: AGGREGATION_TYPE.VALUES_LIST,
+  [AGGERGATION_WIDGET_TYPE.PIE]: AGGREGATION_TYPE.COUNT_PER_VALUE,
+  [AGGERGATION_WIDGET_TYPE.LIKERT_SCALE_AND_PIE]:
+    AGGREGATION_TYPE.LIKERT_SCALE_AND_PIE_AGG,
+  [AGGERGATION_WIDGET_TYPE.REPORTS_BY_TASK]: AGGREGATION_TYPE.REPORTS_BY_TASK,
+  [AGGERGATION_WIDGET_TYPE.REPORTS_MAP]: AGGREGATION_TYPE.OBJECTS_LIST,
+  [AGGERGATION_WIDGET_TYPE.IQR_BOX_PLOT]: AGGREGATION_TYPE.NUMBERS_LIST,
+  [AGGERGATION_WIDGET_TYPE.CALENDAR]: AGGREGATION_TYPE.COUNT_PER_DATE,
+  [AGGERGATION_WIDGET_TYPE.DEFAULT]: AGGREGATION_TYPE.VALUES_LIST
+}
+
+const AGGREGATION_TYPE_FUNCTION = {
+  [AGGREGATION_TYPE.REPORTS_BY_TASK]: reportsByTaskAggregation,
+  [AGGREGATION_TYPE.COUNT_PER_VALUE]: countPerValueAggregation,
+  [AGGREGATION_TYPE.COUNT_PER_DATE]: countPerDateAggregation,
+  [AGGREGATION_TYPE.NUMBERS_LIST]: numbersListAggregation,
+  [AGGREGATION_TYPE.VALUES_LIST]: valuesListAggregation,
+  [AGGREGATION_TYPE.OBJECTS_LIST]: objectsListAggregation,
+  [AGGREGATION_TYPE.LIKERT_SCALE_AND_PIE_AGG]: likertScaleAndPieAggregation
+}
+
+export const getAggregationWidget = (
+  fieldConfig,
+  defaultWidgetPerFieldType = DEFAULT_AGGREGATION_WIDGET_PER_FIELD_TYPE,
+  ignoreFieldConfigWidget = false
+) => {
+  const widget = !ignoreFieldConfigWidget && fieldConfig.aggregation?.widget
+  const defaultWidget = defaultWidgetPerFieldType[fieldConfig.type]
+  const defaultWidgetIsObject = typeof defaultWidget === "object"
+  return (
+    widget ||
+    (defaultWidget &&
+      defaultWidgetIsObject &&
+      defaultWidget[fieldConfig.widget]) ||
+    (!defaultWidgetIsObject && defaultWidget) ||
+    null
+  )
+}
+
+const getAggregationFunction = (fieldConfig, aggregationWidget) => {
+  const aggregationType =
+    fieldConfig.aggregation?.aggregationType ||
+    DEFAULT_AGGREGATION_TYPE_PER_WIDGET_TYPE[aggregationWidget]
+  return aggregationType
+    ? AGGREGATION_TYPE_FUNCTION[aggregationType] || null
+    : null
```

Comment:
Could just be `return AGGREGATION_TYPE_FUNCTION[aggregationType]` as `arr[null] === undefined`.